client/blocks/upsell-nudge/test/index.jsx

```diff
@@ -12,25 +15,34 @@ import {
 	PLAN_JETPACK_BUSINESS,
 	PLAN_JETPACK_BUSINESS_MONTHLY,
 } from '@automattic/calypso-products';
-import { shallow } from 'enzyme';
+import uiReducer from 'calypso/state/ui/reducer';
+import { renderWithProvider } from 'calypso/test-helpers/testing-library';
 import { UpsellNudge } from '../index';
 
-const props = {
-	callToAction: null,
-	title: 'banner title',
-	forceDisplay: true,
-};
+function renderWithRedux( ui ) {
+	return renderWithProvider( ui, {
+		reducers: {
+			ui: uiReducer,
+		},
+	} );
+}
 
 describe( 'UpsellNudge should render a Banner with a class name corresponding to appropriate plan', () => {
+	const props = {
+		callToAction: null,
+		title: 'banner title',
+		forceDisplay: true,
+	};
+
 	[
 		PLAN_PERSONAL,
 		PLAN_PERSONAL_2_YEARS,
 		PLAN_JETPACK_PERSONAL,
 		PLAN_JETPACK_PERSONAL_MONTHLY,
 	].forEach( ( plan ) => {
 		test( 'Personal', () => {
-			const comp = shallow( <UpsellNudge { ...props } plan={ plan } /> );
-			expect( comp.find( '.is-upgrade-personal' ) ).toHaveLength( 1 );
+			const { container } = renderWithRedux( <UpsellNudge { ...props } plan={ plan } /> );
+			expect( container.firstChild ).toHaveClass( 'is-upgrade-personal' );
```

Comment:
Regarding the firstChild approach, is this another good opportunity to use role instead?

Reply:
Nope. Actually, `firstChild` is recommended by the library, see https://testing-library.com/docs/react-testing-library/api/#container-1 so I prefer going with it.

Reply:
Thanks for the link!